src/VecSim/types/bfloat16.h

```diff
@@ -11,8 +11,12 @@
 #include <cmath>
 
 namespace vecsim_types {
-
-using bfloat16 = unsigned short;
+struct bfloat16 {
+    uint16_t val;
+    constexpr bfloat16(uint16_t val = 0) : val(val) {}
```

Comment:
is it beneficial to have an "undefined" variant for declaration only? (like `int x;`)

Reply:
Since 0 is the default value It will be initialized to 0